students/ChrisH/lesson06/mailroom.py

```diff
@@ -0,0 +1,180 @@
+
+            Warmest Regards,
+                Local Charity
+"""
+    amount = donor_data[donor][-1]
+    return format_string.format(last_donation=float(amount), first_name=donor.first, last_name=donor.last)
+
+
+def send_letters_all():
+    """
+        Runs through donor data structure, generates a thank you letter for each and saves it to
+        the current working directory with in a date+donorname.txt file
+        :return: None
+        """
+    for donor in donor_data:
+        print(f'Generating letter for {get_donor_fullname(donor)}')
+        now = time.localtime()
+        f_name = f"{now.tm_year}{now.tm_mon:02}{now.tm_mday:02}_"
+        f_name += get_donor_fullname(donor).replace(" ", "_") + ".txt"
+        with open(f_name, 'w') as file_out:
+            file_out.write(generate_letter(donor))
+    return None
+
+
+def get_donor(fullname):
+    '''
+    Given a text string with a donor's full name, determines if it is in data, new, or invalid.
+    If new, creates new donor entry in donor_data, with an empty donation list.
+    :param fullname: string, full name of a donor
+    :return: Donor object, or None
+    '''
+    name_sp = fullname.split()
+    if len(name_sp) == 0:
+        return None
+    elif len(name_sp) > 1:
+        donor = Donor(name_sp[0], ' '.join(name_sp[1:]))  # Defines first name up to the first space given
+    else:
+        donor = Donor(first='', last=name_sp[0])
+
+    if donor not in donor_data:
+        donor_data[donor] = []
+
+    return donor
+
+
+def send_thank_you():
+    """
+    Prompts for donor name, if not present, adds user to data. Prompts for donation
+    and adds it to donor's data. Prints a 'Thank You' email populated with the donor's data.
+    :return: None
+    """
+
+    while True:
+        name = input("Enter a Full Name ('list' to show list of donors, 'q' to quit): ")
+        if name == 'q':
+            return
+        elif name == 'list':
+            print(("{}\n" * len(donor_data)).format(*([get_donor_fullname(d) for d in donor_data])))
+            continue
+        else:
+            donor = get_donor(name)
+            if not donor:
+                print("Name cannot be empty.")
+                continue
+            else:
+                break
+
+    while True:
+        try:
+            amount = input("Enter a donation amount for {} : ".format(get_donor_fullname(donor)))
+            if float(amount) <= 0:
+                print('Amount donated must be a positive number.')
+            else:
+                break
+        except ValueError:
+            print('Please enter a numerical value.')
+
+    donor_data[donor].append(float(amount))
+    print(generate_letter(donor))
+
+
+def print_report():
+    """
+    Prints a formatted report on the donors with name, amount given, number of gifts, and average gift.
+    :return: None
+    """
+    # Find longest name in donor list, or use name_min value
+    name_min = 25
```

Comment:
you don't use these anywhere else except when setting name_max, just put the value in there, no need for extra variable